### PR TITLE
nixos/tests/zfs: fix using wrong package in tests; minor cleanup

### PR DESCRIPTION
**Pull request by amarshall**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by adamcstephens** [comment]
@ofborg test zfs

**Comment by adamcstephens** [comment]
Looks good on aarch64

```
─❯ nix-build -A nixosTests.zfs
/nix/store/p6cl31i5fzn0p8i0bmg62gdhcxml6lbw-vm-test-run-multi-disk-zfs
/nix/store/0a482yqh9f5z1b286gk2g6lzphcsf4jl-vm-test-run-installer-zfs-root
/nix/store/w8jkjywbfmjv3amvml1sp0g0d5n4dmcv-vm-test-run-installer-separateBootZfs
/nix/store/3jy87dp12s4f7vaxvh4qybrl0h9r56i0-vm-test-run-zfs-2.1-series
/nix/store/8s8px3h1pzqq7jrp12hcszbfffapcn9j-vm-test-run-zfs-stable
/nix/store/jxhjwqdby8khqnqi3fz1bvgbjxa8gcfb-vm-test-run-zfs-unstable
/nix/store/x07rv6rfs898l3mwwhmc77lz8y6nlsv5-vm-test-run-zfs-unstable
```

**Review by adamcstephens — APPROVED** [review]

File: nixos/tests/zfs.nix [hunk 1]
```diff
@@ -8,12 +8,9 @@ with import ../lib/testing-python.nix { inherit system pkgs; };
 let
 
   makeZfsTest = name:
-    { kernelPackage ? if enableUnstable
-                      then pkgs.zfsUnstable.latestCompatibleLinuxPackages
-                      else pkgs.linuxPackages
-    , enableUnstable ? false
+    { kernelPackage
```
Comment (RaitoBezarius) [inline nixos/tests/zfs.nix:11]:
```suggestion
    { kernelPackages
```

Comment (RaitoBezarius) [reply nixos/tests/zfs.nix:11]: kernelPackage should be the ZFS kernel package itself, not the package set.

**Pull request merged**